pkg/ingester/errors.go

```diff
@@ -135,3 +136,95 @@ func formatMaxMetadataPerMetricError(limits *validation.Overrides, labels labels
 	))
 	return makeMetricLimitError(labels, err)
 }
+
+type ingesterErrSamplers struct {
+	sampleTimestampTooOld             *log.Sampler
+	sampleTimestampTooOldOOOEnabled   *log.Sampler
+	sampleTimestampTooFarInFuture     *log.Sampler
+	sampleOutOfOrder                  *log.Sampler
+	sampleDuplicateTimestamp          *log.Sampler
+	maxSeriesPerMetricLimitExceeded   *log.Sampler
+	maxMetadataPerMetricLimitExceeded *log.Sampler
+	maxSeriesPerUserLimitExceeded     *log.Sampler
+	maxMetadataPerUserLimitExceeded   *log.Sampler
+}
+
+func newIngesterErrSamplers(freq int64) ingesterErrSamplers {
+	if freq == 0 {
+		return ingesterErrSamplers{}
+	}
+	return ingesterErrSamplers{
+		sampleTimestampTooOld:             log.NewSampler(freq),
+		sampleTimestampTooOldOOOEnabled:   log.NewSampler(freq),
+		sampleTimestampTooFarInFuture:     log.NewSampler(freq),
+		sampleOutOfOrder:                  log.NewSampler(freq),
+		sampleDuplicateTimestamp:          log.NewSampler(freq),
+		maxSeriesPerMetricLimitExceeded:   log.NewSampler(freq),
+		maxMetadataPerMetricLimitExceeded: log.NewSampler(freq),
+		maxSeriesPerUserLimitExceeded:     log.NewSampler(freq),
+		maxMetadataPerUserLimitExceeded:   log.NewSampler(freq),
+	}
+}
+
+func (i ingesterErrSamplers) WrapSampleTimestampTooOldError(err error) error {
+	if i.sampleTimestampTooOld == nil {
+		return err
+	}
+	return i.sampleTimestampTooOld.WrapError(err)
+}
```

Comment:
We don't need these methods. Since `nil` value of `*log.Sampler` would just `return err`, the caller can safely just call `samplers.sampleTimestampTooOld.WrapError(err)`, and thus there's no need to test this struct at all.

Reply:
Thank you for pointing this out.